homeassistant/components/sonos/config_flow.py

```diff
@@ -43,7 +45,7 @@ async def async_step_zeroconf(
             discovery_manager.async_discovered_player(
                 "Zeroconf", properties, host, uid, boot_seqnum, model, mdns_name
             )
-        return await self.async_step_discovery(discovery_info)
+        return await self.async_step_discovery(cast(dict, discovery_info))
```

Comment:
I think that it would be better to explicitely keep only the required elements but I think it is out of scope for this PR.